query-engine/core/src/query_document/mod.rs

```diff
@@ -159,12 +162,21 @@ pub struct CompactedDocument {
     pub nested_selection: Vec<String>,
     pub operation: Operation,
     pub keys: Vec<String>,
+    pub original_query_options: crate::QueryOptions,
     name: String,
 }
 
 impl CompactedDocument {
+    pub fn throw_on_empty(&self) -> bool {
+        self.original_query_options.contains(QueryOption::ThrowOnEmpty)
+    }
+
     pub fn single_name(&self) -> String {
-        format!("findUnique{}", self.name)
+        if self.throw_on_empty() {
+            format!("findUnique{}OrThrow", self.name)
+        } else {
+            format!("findUnique{}", self.name)
+        }
```

Comment:
Would be great if we could use the `schema::IdentifierType` enum here, but we can't really because we don't have access to the model. We could make that work by using a dyn trait, but it's not worth it atm. Just some out loud thoughts.